src/main/resources/templates/subjectSelection.html

```diff
@@ -0,0 +1,159 @@
+            </div>
+        </div>
+    </div>
+
+    <div class="modal fade" id="errorModal" tabindex="-1" role="dialog" aria-labelledby="errorModalLabel" aria-hidden="true">
+        <div class="modal-dialog" role="document">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h5 class="modal-title" id="errorModalLabel">Warning!</h5>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                </div>
+                <div class="modal-body" id="errorMessage">
+
+                </div>
+                <div class="modal-footer">
+                    <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
+
+<script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.7/umd/popper.min.js"></script>
+<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js"></script>
+
+
+<script>
+    $(document).ready(function() {
+        $('#addBtn').click(function() {
+            var selectedSubject = $('#subjects option:selected');
+            var selectedSubjectText = selectedSubject.text();
+            var selectedSubjectValue = selectedSubject.val();
+
+            var selectedYear = $('#schoolYear option:selected');
+            var selectedYearText = selectedYear.text();
+            var selectedYearValue = selectedYear.val();
+
+            var combinedValue = `${selectedSubjectValue}-${selectedYearValue}`;
+
+            $(`.selected-subject[data-combined-value="${combinedValue}"]`).remove();
+
+            if (selectedSubjectText && selectedYearText) {
+                var subjectHtml = `
+            <div class="selected-subject" data-combined-value="${combinedValue}">
+                <span>${selectedSubjectText} - ${selectedYearText}</span>
+                <button type="button" class="btn btn-link remove-btn"><i class="fas fa-trash"></i></button>
+                <input type="hidden" name="selectedValues" value="${combinedValue}">
+            </div>`;
+                $('#selectedSubjectsContainer').append(subjectHtml);
+            }
+        });
+
+        $('#selectedSubjectsContainer').on('click', '.remove-btn', function () {
+            $(this).closest('.selected-subject').remove();
+        });
+
+        $('#submitFormBtn').click(function () {
+            var formData = $("#subjectForm").serialize();
+            console.log("submit-btn click");
+
+            $.ajax({
+                type: "POST",
+                url: "/userSubject/save",
+                data: formData,
+                success: function (response) {
+                    console.log(response);
+                    if (response.success) {
```

Comment:
then comes error message

Reply:
no, this is not correct. You only check in the success function wheter the value is true, then you navigate, otherwise you are doing nothing in the success function

Reply:
I didn't understand what I have to fix here, shall I remove success?

Reply:
in your controller you have: 
- success cases; they set "success" to true, and return a success status code (eg 200)
- error cases: they set "success" to false and return an error status code (eg 400, 404, 500)

but in your view you only check "success" value when the response code already is successfull; in all error cases only the error function will be called and not the success function. so there is no need to have the success value at all, because you work with status codes to determine success. 

all in all the 'if (response.success)' is useless, because it will always be true with your implementation. and if it not be true, there would not be any indication for the user as well. so yes, I would suggest removing it here and in the controller

Reply:
many thanks for explanation